cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -287,3 +288,78 @@ fn storage_footprint_to_ledger_footprint(foot: &Footprint) -> Result<LedgerFootp
         read_write: read_write.try_into()?,
     })
 }
+
+pub(crate) fn compute_bump_footprint_exp_transaction_data_and_min_fee(
+    footprint: LedgerFootprint,
+    _ledgers_to_expire: u32,
+    snapshot_source: &ledger_storage::LedgerStorage,
+) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
+    let read_bytes =
+        calculate_unmodified_ledger_entry_bytes(footprint.read_only.as_vec(), snapshot_source)?;
+    let soroban_resources = SorobanResources {
+        // TODO: is this correct or should we use the footprint?
+        footprint: LedgerFootprint {
```

Comment:
The request should contain the list of keys to bump and `ledgers_to_expire` and this should just copy the key list to RO footprint.

Reply:
I will just pass on the footprint then, let me know if that's what you meant

Reply:
This would work, but I would suggest to accept a list of the ledger keys from the user, so that they don't need to worry about RO vs RW.

Reply:
@dmkozh I don't see the user really being concerned about that. In the context of `simulateTransaction`, I think that this is the correct approach.
The user is likely to be using the sdks to use these features, and that functionality would be implemented there.
Depending on the usage, we might want to create a dedicated API for that sole purpose, but other then cleaning up some of the REST interface, the won't be much gains there.

Reply:
Well, the design is really up to you. Semantically these operations just bump/restore a set of keys and ideally their interface would be to just take a list of keys. I just want to make it clear that the way they're implemented in XDR is indeed confusing, but that seemed like a lesser evil out of the alternatives. Here you have more freedom, so you could provide a more semantically correct interface. It also would be more consistent with invoke fn (where user passes the input data and gets the footprint back).